include/cuco/detail/bloom_filter/bloom_filter_impl.cuh

```diff
@@ -152,10 +152,10 @@ class bloom_filter_impl {
 
 #pragma unroll
       for (uint32_t i = rank; i < optimal_num_threads; i += num_threads) {
-        auto const word = policy_.word_pattern(hash_value, rank);
+        auto const word = policy_.word_pattern(hash_value, i);
 
         auto atom_word =
-          cuda::atomic_ref<word_type, thread_scope>{*(words_ + (idx * words_per_block + rank))};
+          cuda::atomic_ref<word_type, thread_scope>{*(words_ + (idx * words_per_block + i))};
```

Comment:
Ha, very good catch! 🎣 The bug would have surface when we use a CG that is smaller than the optimal CG size.